NEWS.md

```diff
@@ -4,11 +4,17 @@
 
 ### Breaking changes
 
+- Several functions have been rewritten to match the behavior of Python Polars.
+  - `pl$implode(...)` is rewritten to be a syntactic sugar for `pl$col(...)$implode()` (#923).
+  - Unify names of input/output function arguments (935).
+    - All arguments except the first argument must be named arguments.
+    - In `pl$read_*` and `pl$scan_*` functions, the first argument is now `source`.
+    - In `<DataFrame>$write_*` functions, the first argument is now `file`.
+    - In `<LazyFrame>$sink_*` functions, the first argument is now `path`.
```

Comment:
Good point.

I believe that Python uses different names by design because these two are of different classes.
`str | Path | IO[str] | IO[bytes]` v.s. `str | Path`
https://github.com/pola-rs/polars/blob/ec8f5e3a414c857a65bbb3db83720275282b14b4/py-polars/polars/dataframe/frame.py#L2579
https://github.com/pola-rs/polars/blob/ec8f5e3a414c857a65bbb3db83720275282b14b4/py-polars/polars/lazyframe/frame.py#L2282

I think this is important in terms of compatibility with common file operations using the IO module.
https://docs.python.org/3/library/io.html

In the R world, it may not be worthwhile to make a distinction between these, so we may as well unify them into `file`.

Reply:
Yes I think having a single `file` arg is better.

Reply:
Sorry, I checked again carefully and this is something else at the Rust level, so I still think another argument name makes sense.

Reply:
I don't understand this argument. I think there's no point in having `file` for write functions and `path` for sink functions so we should choose one, and the name given on the Rust side shouldn't matter for this choice.

>  I think having a single file arg is better.

Actually `path` should be preferred since those write/sink functions might be able to create hive partitions

Reply:
> the name given on the Rust side shouldn't matter for this choice.

Since these are actually implemented differently at the Rust level, I think we need to respect the variable names on the Rust side.
https://docs.pola.rs/user-guide/io/csv/